app/webpacker/components/ContactsPage/SubForms/Competition/index.jsx

```diff
@@ -1,23 +1,64 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { FormTextArea } from 'semantic-ui-react';
+import { useQuery } from '@tanstack/react-query';
 import WcaSearch from '../../../SearchWidget/WcaSearch';
 import SEARCH_MODELS from '../../../SearchWidget/SearchModel';
 import I18n from '../../../../lib/i18n';
+import useQueryParams from '../../../../lib/hooks/useQueryParams';
+import { fetchJsonOrError } from '../../../../lib/requests/fetchWithAuthenticityToken';
+import { apiV0Urls } from '../../../../lib/requests/routes.js.erb';
+import Loading from '../../../Requests/Loading';
+import { itemToOption } from '../../../SearchWidget/MultiSearchInput';
 
-export default function Competition({ formValues, setFormValues }) {
+const FORM_DEFAULT_VALUE = {
+  competition: null,
+  message: '',
+};
+
+export default function Competition({ setSubformValues, setFormValid }) {
+  const [queryParams] = useQueryParams();
+  const [formValues, setFormValues] = useState(FORM_DEFAULT_VALUE);
   const handleFormChange = (_, { name, value }) => setFormValues(
     { ...formValues, [name]: value },
   );
+  const { data: competitionData, isLoading } = useQuery({
+    queryKey: ['competition'],
+    queryFn: () => fetchJsonOrError(apiV0Urls.competitions.info(queryParams.competitionId)),
+    enabled: !!queryParams?.competitionId,
+  });
+
+  useEffect(() => {
+    setSubformValues(formValues);
+    setFormValid(formValues.competition !== null && formValues.message?.length > 0);
```

Comment:
This should not be done by manually computing your booleans. SemUI Form elements have a `required` property which does most* of the work for you

*Exception: Checkboxes are a little bit glitched out. Those are the only exception where we have to compute `isValid` booleans ourselves because of a glitch in the SemUI React integration. Everything else should be solvable by just setting the input fields to `required` and calling it a day.

If you can come up with an argument why it's really, _really_ required to do those validations yourself, you should probably do them on the top level and switch business logic based on which `contactRecipient` is selected.